### DISCUSSION TITLE
How push Middleware to application from a Filament package?

### DISCUSSION THREAD

**Discussion by dasundev** [filamentphp/filament#6849] [Help]
I tried to push a middleware from my plugin service provider. But it doesn't work. Can anyone explain the problem here?

```php
public function bootingPackage(): void
{
    $this->app[Kernel::class]->prependMiddleware(MyPluginMiddleware::class);
}
```

**Comment by danharrin** [comment]
You cant do this from a v2 plugin, you just need to get your users to add the middleware to their config/filament.php

**Reply by dasundev** [reply]
Oh Okay! 

I've used `Config::prepend('filament.middleware.base', MyPluginMiddleware::class)` 

Instead of getting users to add the middleware to their config/filament.php manually, this way does the same job automatically.

I believe it would be great, @danharrin what do you think?

**Reply by danharrin** [reply]
As long as your users have the option to disable it and register their own order if they want

**Reply by dasundev** [reply]
I will hold this conversation for now and definitely will add a reply here after the package is published on Filament. then we can continue the discussion. 😉

**Reply by dasundev** [reply]
@danharrin Why I can not read a cookie within the middleware? I have added my middleware to the end of `middleware.base` array at config/filament.php
